src/ApparentHorizons/StrahlkorperGr.cpp

```diff
@@ -449,62 +499,81 @@ tnsr::ii<DataVector, 3, Frame> extrinsic_curvature(
     }
   }
 
-  auto extrinsic_curvature(grad_normal);
-
+  *result = grad_normal;
   for (size_t i = 0; i < 3; ++i) {
     for (size_t j = i; j < 3; ++j) {
-      extrinsic_curvature.get(i, j) += unit_normal_one_form.get(i) *
-                                       unit_normal_one_form.get(j) *
-                                       get(nI_nJ_gradnij);
+      result->get(i, j) += unit_normal_one_form.get(i) *
+                           unit_normal_one_form.get(j) * get(nI_nJ_gradnij);
       for (size_t k = 0; k < 3; ++k) {
-        extrinsic_curvature.get(i, j) -=
+        result->get(i, j) -=
             unit_normal_vector.get(k) *
             (unit_normal_one_form.get(i) * grad_normal.get(j, k) +
              unit_normal_one_form.get(j) * grad_normal.get(i, k));
       }
     }
   }
-  return extrinsic_curvature;
 }
 
 template <typename Frame>
-Scalar<DataVector> ricci_scalar(
+tnsr::ii<DataVector, 3, Frame> extrinsic_curvature(
+    const tnsr::ii<DataVector, 3, Frame>& grad_normal,
+    const tnsr::i<DataVector, 3, Frame>& unit_normal_one_form,
+    const tnsr::I<DataVector, 3, Frame>& unit_normal_vector) noexcept {
+  tnsr::ii<DataVector, 3, Frame> result{};
+  extrinsic_curvature(make_not_null(&result), grad_normal, unit_normal_one_form,
+                      unit_normal_vector);
+  return result;
+}
+
+template <typename Frame>
+void ricci_scalar(
+    const gsl::not_null<Scalar<DataVector>*> result,
     const tnsr::ii<DataVector, 3, Frame>& spatial_ricci_tensor,
     const tnsr::I<DataVector, 3, Frame>& unit_normal_vector,
     const tnsr::ii<DataVector, 3, Frame>& extrinsic_curvature,
     const tnsr::II<DataVector, 3, Frame>& upper_spatial_metric) noexcept {
-  auto ricci_scalar = trace(spatial_ricci_tensor, upper_spatial_metric);
+  destructive_resize_components(result, spatial_ricci_tensor.begin()->size());
```

Comment:
shouldn't need if mutating version calls `destructive_resize_components` or if there is not a mutating version